### PR TITLE
Adjunctions

### PR DESCRIPTION
**Pull request by sjoerdvisscher**

### CODE REVIEW

File: idris2/Profunctors/HomFunctor.idr [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  (\a => mor cat (fst a) (snd a))
+  (\a, b, f => MkExtensionalTypeMorphism (\h => compose cat (fst b) (fst a) (snd b) (pi1 f) (compose cat (fst a) (snd a) (snd b) h (pi2 f))))
+  (\a => funExt (\x => trans (leftIdentity cat (fst a) (snd a) _) (rightIdentity cat (fst a) (snd a) x)))
+  (\a, b, c, f, g => funExt (\x =>
+    trans (sym (associativity cat _ _ _ _ (pi1 g) (pi1 f) _))
+    (cong (compose cat (fst c) (fst b) (snd c) (pi1 g)) (
+      trans (cong (compose cat (fst b) (fst a) (snd c) (pi1 f)) (associativity cat _ _ _ _ x (pi2 f) (pi2 g)))
+      (associativity cat _ _ _ _ (pi1 f) _ (pi2 g))))))
+
+-- costar f takes f : C -> D to D(F, 1)
```
Comment (marcosh) [inline idris2/Profunctors/HomFunctor.idr]: should `F` here be `f`? Same below

File: src/Profunctors/HomFunctor.lidr [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+>   (\a => mor cat (fst a) (snd a))
+>   (\a, b, f => MkExtensionalTypeMorphism (\h => compose cat (fst b) (fst a) (snd b) (pi1 f) (compose cat (fst a) (snd a) (snd b) h (pi2 f))))
+>   (\a => funExt (\x => trans (leftIdentity cat (fst a) (snd a) _) (rightIdentity cat (fst a) (snd a) x)))
+>   (\a, b, c, f, g => funExt (\x =>
+>     trans (sym (associativity cat _ _ _ _ (pi1 g) (pi1 f) _))
+>     (cong (
+>       trans (cong (associativity cat _ _ _ _ x (pi2 f) (pi2 g)))
+>       (associativity cat _ _ _ _ (pi1 f) _ (pi2 g))))))
+
+costar f takes f : C -> D to D(F, 1)
```
Comment (marcosh) [inline src/Profunctors/HomFunctor.lidr]: same comment as above, should `F` be `f` here? Same below

File: idris2/Profunctors/HomFunctor.idr [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import Basic.Functor
+import Dual.DualCategory
+import Dual.DualFunctor
+import Idris.TypesAsCategoryExtensional
+import Product.ProductCategory
+import Product.ProductFunctor
+import Profunctors.Profunctor
+
+public export
+homFunctor : (cat : Category) -> Profunctor cat cat
```
[the remark this answers is not in the document]
Comment (sjoerdvisscher) [reply idris2/Profunctors/HomFunctor.idr]: Nobody calls this the hom profunctor, it is always called the hom functor. A profunctor is still a functor anyways.

[the remark this answers is not in the document]
Comment (marcosh) [reply idris2/Profunctors/HomFunctor.idr]: ok, lets's keep it as it is

**Pull request merged**